File: packages/gutenberg/android/src/main/java/org/wpnative/gutenberg/WPNativeGutenbergModule.kt
// Copyright (C) 2026 Chris Huber
// SPDX-License-Identifier: GPL-2.0-or-later

package org.wpnative.gutenberg

import expo.modules.kotlin.Promise
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class WPNativeGutenbergModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("WPNativeGutenberg")

        View(GutenbergEditorView::class) {
            Prop("initialTitle") { view: GutenbergEditorView, title: String ->
                view.setInitialTitle(title)
            }

            Prop("initialContent") { view: GutenbergEditorView, content: String ->
                view.setInitialContent(content)
            }

            Events("onReady", "onError")

            OnViewDidUpdateProps { view: GutenbergEditorView ->
                view.loadEditorIfNeeded()
            }

            AsyncFunction("requestContent") { view: GutenbergEditorView, promise: Promise ->
                view.requestContent(promise)
            }
        }
    }
}

File: packages/gutenberg/ios/WPNativeGutenbergModule.swift
// Copyright (C) 2026 Chris Huber
// SPDX-License-Identifier: GPL-2.0-or-later

import ExpoModulesCore

public final class WPNativeGutenbergModule: Module {
  public func definition() -> ModuleDefinition {
    Name("WPNativeGutenberg")

    View(GutenbergEditorView.self) {
      Prop("initialTitle") { (view: GutenbergEditorView, title: String) in
        MainActor.assumeIsolated {
          view.initialTitle = title
        }
      }

      Prop("initialContent") { (view: GutenbergEditorView, content: String) in
        MainActor.assumeIsolated {
          view.initialContent = content
        }
      }

      Events("onReady", "onError")

      OnViewDidUpdateProps { (view: GutenbergEditorView) in
        MainActor.assumeIsolated {
          view.loadEditorIfNeeded()
        }
      }

      AsyncFunction("requestContent") { (view: GutenbergEditorView) async throws in
        try await view.requestContent()
      }
    }
  }
}
